### ISSUE TITLE
Error while trying to find Git in reuse lint

### ISSUE THREAD

**Issue by SachaHq** [fsfe/reuse-tool#195]
Hello,

I am using reuse as a user on Windows 10 and I had the following issue when using the `reuse lint` command:
1. A message indicated *reuse.project - WARNING - could not find Git* and a lot of files that are supposed to be excluded from the lint inspection because part of .gitignore were not ignored. I assumed this was because of reuse not finding Git.
2. I added Git path to the Path (and I also tried changing Git location) because I had it on a different disk, and I assumed that the `shutil.which` function that is used to define *GIT_EXE* in the *_utils* code could not find it for this reason. Then I had the following error:  
`Traceback (most recent call last):`  
 ` File "c:\python3.6.0\lib\runpy.py", line 193, in _run_module_as_main`  
`    "__main__", mod_spec)`  
 ` File "c:\python3.6.0\lib\runpy.py", line 85, in _run_code`  
  `  exec(code, run_globals)`  
 ` File "C:\Users\myusername\AppData\Roaming\Python\Python36\Scripts\reuse.exe\__main__.py", line 7, in <module>`  
 ` File "C:\Users\myusername\AppData\Roaming\Python\Python36\site-packages\reuse\_main.py", line 247, in main`  
`    project = create_project()`  
`  File "C:\Users\myusername\AppData\Roaming\Python\Python36\site-packages\reuse\project.py", line 322, in create_project`  
 `   root = find_root()`  
`  File "C:\Users\myusername\AppData\Roaming\Python\Python36\site-packages\reuse\_util.py", line 95, in find_root`  
 `   if in_git_repo(cwd):`  
`  File "C:\Users\myusername\AppData\Roaming\Python\Python36\site-packages\reuse\_util.py", line 140, in in_git_repo`  
  `  result = execute_command(command, _LOGGER, cwd=cwd)`  
 ` File "C:\Users\myusername\AppData\Roaming\Python\Python36\site-packages\reuse\_util.py", line 86, in execute_command`  
`    command, stdout=stdout, stderr=stderr, check=False, **kwargs`  
`  File "c:\python3.6.0\lib\subprocess.py", line 403, in run`  
   ` with Popen(*popenargs, **kwargs) as process:`  
 ` File "c:\python3.6.0\lib\subprocess.py", line 707, in __init__`  
 `   restore_signals, start_new_session)`  
 ` File "c:\python3.6.0\lib\subprocess.py", line 990, in _execute_child`  
`    startupinfo)`  
`TypeError: CreateProcess() argument 8 must be str or None, not WindowsPath`  

I am assuming this is because I am using Windows and not Linux. If you have any clue of how to solve this issue or of what I did wrong, I would be very grateful. Do not hesitate to ask if you need any information I did not mention.

Thank you for your kind help and your good work on *Reuse*

Sacha

**Comment by carmenbianca** [comment]
Hi @SachaHq 

I'm a little short on time for contributions to REUSE at the moment, but I can identify the problem:

```
TypeError: CreateProcess() argument 8 must be str or None, not WindowsPath
```

This is strange, and points to a bug in Python. You can try updating to Python 3.8 to fix this.

If that doesn't work, it should be fairly simple to fix this downstream by converting the path to a string. Please let me know if updating to Python 3.8 works for you, though.

**Comment by SachaHq** [comment]
Dear @carmenbianca ,

Thank you very much for your quick reply. I did update to Python 3.8 but it did not solve this issue:
- It still seems that `shutil.which` does not find *git*, 
- When it does (when I move git.exe in the very project directory), I still have the same error message.  

Since it is not due to the Python version, perhaps it is the *GIT_EXE* definition in *_util* that is not recognised afterwards because it is a WindowsPath ?
> GIT_EXE = shutil.which("git")

Do not hesitate if you find a way to fix this.

Thanks again

**Comment by carmenbianca** [comment]
Okay, trying to decipher this because some things don't make sense to me.

1. GIT_EXE (`git`) not being recognised by `shutil.which` is a case of having to update the Windows PATH. Or maybe it should say `Git.exe`? I have no idea, and I have no Windows machine to test this on.

2. This appears to be a Windows-specific bug after all.

For (2.), I assume that the following would fail under Windows:

```python
import pathlib
import subprocess

subprocess.run(["git"], cwd=pathlib.Path.cwd())
```

^ This succeeds under Linux. If it does fail on Windows, please create a bug report at https://bugs.python.org/

I can fix this fairly easily, though. Will incorporate it into #197

**Comment by SachaHq** [comment]
Dear @carmenbianca, 

Sorry if I am not clear enough in my messages.

1. To explain a little bit further: 
   a. when *git.exe* is in another directory and not in the Windows PATH, no error occur with reuse but it is specified that git was not found, so important functions such as the *.gitignore* recognition are not taken into account. 
   b. When I add *git.exe* to the directory, or its path to the Windows PATH, then the error I mentioned occurs. 
That is why I assumed the issue came from the `shutil.which` not finding git in the a. case, or finding it but with an unrecognisable format for Python on Windows in the b. case
2. Good assumption: this does fail under Windows, with the same error  I had previously:
[...]`TypeError: CreateProcess() argument 8 must be str or None, not WindowsPath`

Great news if it is being incorporated.

Thanks again for your time and for your work

**Comment by carmenbianca** [comment]
> Sorry if I am not clear enough in my messages.

No worries! Thank you for filing the bug, and you were more than clear: It's just super difficult to debug things when I haven't used Windows in 10 years :p 

> When I add git.exe to the directory, or its path to the Windows PATH

Having to add the git binary to PATH seems normal. I might add a note about this to the documentation.

> Good assumption: this does fail under Windows, with the same error I had previously

Could you please make a bug report about this on https://bugs.python.org/ ? You can refer to this issue, and I'll happily back up the bug report if you link it to me.

With kindness,\
Carmen

**Comment by SachaHq** [comment]
Sure, I am doing it right now, thank you for your help !
Sacha

**Comment by SachaHq** [comment]
Dear @carmenbianca ,
While preparing to report the bug to python, I managed to get this to work properly in the end.  
The fact is that the first time I changed my Python version to 3.8 as you suggested, my virtual environment on Pycharm was not properly implemented. Now it is and I can tell you that you were right from the beginning: this was a Python error, and it has been solved between Python 3.6 and Python 3.8. Truly sorry for this little time waste, and thank you again for your quick support.
Kind regards,
Sacha
